File: dotnet/test/MyDotey.NameCacheTest/NameCacheTest.cs
using System;
using Xunit;

using MyDotey.NameCache;
using NC = MyDotey.NameCache.NameCache;

namespace MyDotey.NameCacheTest
{
    public class NameCacheTest
    {
        [Fact]
        public void TestGetName()
        {
            NC nameCache = new NC(".");
            CachedName name1 = nameCache.Get("1", "2", "3");
            CachedName name2 = nameCache.Get("1", "2", "3");

            Assert.True(Object.ReferenceEquals(name1, name2));
            Assert.True(Object.ReferenceEquals(name1.Name, name2.Name));
        }
    }
}

File: java/name-cache/src/test/java/org/mydotey/namecache/NameCacheTest.java
package org.mydotey.namecache;

import org.junit.Assert;
import org.junit.Test;

/**
 * @author koqizhao
 *
 * Jul 20, 2018
 */
public class NameCacheTest {

    @Test
    public void testGetName() {
        NameCache nameCache = new NameCache(".");
        CachedName name1 = nameCache.get("1", "2", "3");
        CachedName name2 = nameCache.get("1", "2", "3");

        Assert.assertTrue(name1 == name2);
        Assert.assertTrue(name1.getName() == name2.getName());
    }

}
